rasterio/fill.py

```diff
@@ -35,8 +36,8 @@ def fillnodata(
         A mask band indicating which pixels to interpolate. Pixels to
         interpolate into are indicated by the value 0. Values > 0
         indicate areas to use during interpolation. Must be same shape
-        as image. If `None`, a mask will be diagnosed from the source
-        data.
+        as image. If None, the inverse of the image's mask will be used
+        if available.
```

Comment:
@sgillies could you rephrase to make this more clear?  E.g., 

```
If `None` and the `image` is a `MaskedArray`, all masked pixels 
(`image.mask==True`) will be interpolated into, and all 
other pixels will be used for interpolation.  
```

Reply:
Though looking at your usage, it looks like anything that has a `mask` attribute that behaves the same as a `MaskedArray` will work.  Not quite sure how to say that in the docstring though.

Reply:
I'll switch to `isinstance(image, MaskedArray)`.